pkg/operator/starter.go

```diff
@@ -481,6 +483,18 @@ func hyperShiftReplaceNamespaceHook(operandNamespace string) dc.DeploymentHookFu
 	}
 }
 
+func hyperShiftSetSecurityContext() dc.DeploymentHookFunc {
+	return func(_ *operatorv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		_, ok := os.LookupEnv("HYPERSHIFT_SUPPORTS_SCC")
+
+		if !ok {
+			deployment.Spec.Template.Spec.SecurityContext.RunAsNonRoot = ptr.To(false)
+			deployment.Spec.Template.Spec.SecurityContext.RunAsUser = ptr.To[int64](1001)
```

Comment:
If the container runs as 1001, why it has `RunAsNonRoot: false` ?

Reply:
And... would the same change work in OCP? I think we could add it to our standalone + hypershift yaml manifests unconditionally.

Reply:
> If the container runs as 1001, why it has RunAsNonRoot: false ?

It looks like these fields are being explicitly set to true in the two containers, listed in the story, that are failing to be created. 
* https://github.com/openshift/cluster-csi-snapshot-controller-operator/blob/ed8a8bcc29ba79e5abedd41deaee9747d0fe14cf/assets/csi_controller_deployment.yaml#L27
* https://github.com/openshift/cluster-csi-snapshot-controller-operator/blob/ed8a8bcc29ba79e5abedd41deaee9747d0fe14cf/assets/webhook_deployment.yaml#L61

Setting them to false behaves the same as not setting them at all per https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.29/#securitycontext-v1-core.

Reply:
> And... would the same change work in OCP? I think we could add it to our standalone + hypershift yaml manifests unconditionally.

I'm not sure what you're asking for here. The current setup is working in OCP today since it supports SCC.

Reply:
>> And... would the same change work in OCP? I think we could add it to our standalone + hypershift yaml manifests unconditionally.

> I'm not sure what you're asking for here. The current setup is working in OCP today since it supports SCC.

It might be better to set UID in yaml files for all OCP flavours instead of introducing a new API (env. var) + go code.

Using explicit UID needs `nonroot` SCC instead of `restricted`, so it's not just adding `runAsUser: 1001`. I'm experimenting in https://github.com/openshift/cluster-csi-snapshot-controller-operator/pull/199, but I'd need to run it by other team members.

Reply:
Would that fix the original issue though in the Jira ticket?
```
Error: container has runAsNonRoot and image will run as root (pod: "csi-snapshot-controller-cfb96bff7-7tc94_clusters-brcox-hypershift-arm(45ab89f2-9c00-4afa-bece-7846505edbfc)", container: snapshot-controller) 
```

With `runAsNonRoot` still set to true, it seems like it would to me. Not 100% sure though.

Reply:
I finally got https://github.com/openshift/cluster-csi-snapshot-controller-operator/pull/199 working, I had to pass the SCC to too many service accounts and it does not look very pretty. Still, it should work the same on Kubernetes and OCP management cluster, at least in theory.

Reply:
Cool. What is the plan? Move ahead with your PR and close this one?

Reply:
Sorry it took so long. We (storage team) agreed that an env. var + code is the lesser evil. We did not find just a yaml file change that would work everywhere.

However, there are issues with your PR, I will submit them as new comments to keep the discussion organized.